src/storage/table_index_list.cpp

```diff
@@ -9,10 +9,11 @@
 #include "duckdb/main/config.hpp"
 
 namespace duckdb {
-void TableIndexList::AddIndex(unique_ptr<Index> index) {
+Index *TableIndexList::AddIndex(unique_ptr<Index> index) {
 	D_ASSERT(index);
 	lock_guard<mutex> lock(indexes_lock);
 	indexes.push_back(std::move(index));
+	return indexes.at(indexes.size() - 1).get();
```

Comment:
this should return a `Index &`
Easiest way is probably to do:
```
auto &result = *index;
indexes.push_back(std::move(index));
return result;
```

Reply:
Fixed.